src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -44,56 +44,58 @@ public static class PackageDownloader
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token))
+            // While CancellationTokenSource has a dispose, if we call it before the tasks using
+            // the token are done we risk having these tasks fail with a object disposed exception so
+            // this should not be in a using unless we want to block until all tasks are done/cancled
+            var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
```

Comment:
@deepakaravindr 
It is ultimately up to you however I vote against option 1 as the only solution.  If we keep the race condition in and only handle it in this location the rest of the code paths still are at risk.  While we can say that the current code won't throw the exception it means all future code in this path will have potential issues.  Tracking down race conditions is always difficult so my preference is to remove it.